eZ/Publish/SPI/MVC/Templating/BaseRenderStrategy.php

```diff
@@ -29,30 +28,28 @@ abstract class BaseRenderStrategy implements RenderStrategy
     protected $requestStack;
 
     public function __construct(
-        iterable $renderMethods,
-        string $defaultMethod,
+        iterable $fragmentRenderers,
+        string $defaultRenderer,
         SiteAccess $siteAccess,
         RequestStack $requestStack
     ) {
-        $this->defaultMethod = $defaultMethod;
-        $this->siteAccess = $siteAccess;
-
-        foreach ($renderMethods as $renderMethod) {
-            $this->renderMethods[$renderMethod->getIdentifier()] = $renderMethod;
+        foreach ($fragmentRenderers as $fragmentRenderer) {
+            $this->fragmentRenderers[$fragmentRenderer->getName()] = $fragmentRenderer;
         }
+
+        $this->defaultRenderer = $defaultRenderer;
+        $this->siteAccess = $siteAccess;
         $this->requestStack = $requestStack;
     }
 
-    protected function getRenderMethod(RenderOptions $options, ValueObject $valueObject): RenderMethod
+    protected function getFragmentRenderer(string $name): FragmentRendererInterface
     {
-        $method = $options->get('method', $this->defaultMethod);
-
-        if (empty($this->renderMethods[$method])) {
-            throw new InvalidArgumentException('method', sprintf(
-                "Method '%s' is not supported for %s.", $method, get_class($valueObject)
-            ));
+        if (!empty($this->fragmentRenderers[$name])) {
+            return $this->fragmentRenderers[$name];
         }
 
-        return $this->renderMethods[$method];
+        throw new InvalidArgumentException('method', sprintf(
+            'Fragment renderer "%s" does not exist.', $name
+        ));
```

Comment:
I actually prefer the original version with exception being thrown as a special condition and return being the result of normal execution flow. This allows new exception conditions to be added down the line, if needed.